optuna/storages/_journal/storage.py

```diff
@@ -108,6 +114,21 @@ def __setstate__(self, state: Dict[Any, Any]) -> None:
         self._replay_result = JournalStorageReplayResult(self._worker_id_prefix)
         self._thread_lock = threading.Lock()
 
+    def restore_replay_result(self, snapshot: bytes) -> None:
+        try:
+            r: Optional[JournalStorageReplayResult] = pickle.loads(snapshot)
+        except (pickle.UnpicklingError, KeyError):
+            _logger.warning("Failed to restore `JournalStorageReplayResult`.")
+            return
+        if r is None:
+            return
+        if not isinstance(r, JournalStorageReplayResult):
+            _logger.warning("The restored object is not `JournalStorageReplayResult`.")
+            return
+        r._worker_id_prefix = self._worker_id_prefix
+        r._worker_id_to_owned_trial_id = {}
```

Comment:
It's optional but we could add `r._last_created_trial_id_by_this_process = -1` for clarity.

Reply:
Thanks for the good catch! I will fix it.